lldb/source/Target/StopInfo.cpp

```diff
@@ -982,6 +982,13 @@ class StopInfoWatchpoint : public StopInfo {
             m_should_stop = false;
           }
         }
+
+        // Don't stop if the watched region value is unmodified, and
+        // this is a Modify-type watchpoint.
+        if (m_should_stop && !wp_sp->WatchedValueReportable(exe_ctx)) {
+          m_should_stop = false;
+        }
```

Comment:
LLVM [says](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements) don’t use braces on simple single-statement bodies.